presto-main/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -1341,4 +1343,46 @@ public Slice next()
             }
         };
     }
+
+    @SqlNullable
+    @Description("Decodes a GMaps-Polyline string into an array of Points")
+    @ScalarFunction("gmaps_polyline_decode")
+    @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
+    public static Block gMapsPolylineDecode(@SqlType(StandardTypes.VARCHAR) Slice polyline) throws PrestoException
+    {
+        final double precision = Math.pow(10.0, (double) DEFAULT_POLYLINE_PRECISION_EXPONENT);
```

Comment:
Does this presume the underlying polyline has this precision?  i.e. if a polyline was encoded using higher precision, would this result in information loss?

Reply:
Yes. The standard and almost all implementations and use cases use 1e5. There are a couple of cases out there that use a precision of 6.

Ordinarily, an optional argument overriding the precision in the function might be worthwhile ([this is how the python library does it](https://pypi.org/project/polyline/)). 

However, please correct me if I'm wrong; so I have specified a separate function that specifies precision.

Reply:
To me this seems like [`approx_percentile`](https://prestodb.io/docs/current/functions/aggregate.html#approx_percentile-x-percentage-same-as-x), which has two forms: one in which you can specify accuracy, and one in which we use a default accuracy and you don't specify it.  I think we can take the same approach here.